test/integration/testpaths/testpath_disable_enable_zone.py

```diff
@@ -0,0 +1,485 @@
+        return
+
+    @attr(tags=["advanced", "basic"],required_hardware=“true”)
+    def test_01_disable_enable_zone(self):
+        """snapshot hardning
+            1. Disable zone and verify following things:
+                For admin user:
+                    1. Should be create to start/stop exsiting vms
+                    2. Should be create to deploy new vm, snapshot,volume,template,iso in the same zone
+                For Non-admin user:
+                    1. Should be create to start/stop exsiting vms
+                    2. Should not be create to deploy new vm, snapshot,volume,template,iso in the same zone
+            2. Enable the above disabled zone and verify that:
+                -All users should be create to deploy new vm, snapshot,volume,template,iso in the same zone
+            3. Try to delete the zone and it should fail with error message:
+                -"The zone is not deletable because there are servers running in this zone"
+
+        """
+        # Step 1
+        vm_user = VirtualMachine.create(
+            self.userapiclient,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+            mode=self.zone.networktype)
+
+        vm_root = VirtualMachine.create(
+            self.apiclient,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+            mode=self.zone.networktype)
+
+        cmd = updateZone.updateZoneCmd()
+        cmd.id = self.zone.id
+        cmd.allocationstate = "Disabled"
+        self.apiclient.updateZone(cmd)
+        zoneList = Zone.list(self.apiclient, id=self.zone.id)
+
+        self.assertEqual(zoneList[0].allocationstate,
+                         "Disabled",
+                         "Check if the zone is in disabled state"
+                         )
+        # Verify the exsisting vms should be running
+        self.assertEqual(vm_user.state,
+                         "Running",
+                         "Verify that the vm is running")
+
+        self.assertEqual(vm_root.state,
+                         "Running",
+                         "Verify that the vm is running")
+
+        time.sleep(30)
+        vm_root.stop(self.apiclient)
+        vm_user.stop(self.apiclient)
+
+        root_state = self.dbclient.execute(
+                "select state from vm_instance where name='%s'" %
+                vm_root.name)[0][0]
+
+        user_state = self.dbclient.execute(
+                "select state from vm_instance where name='%s'" %
+                vm_user.name)[0][0]
+
+        self.assertEqual(root_state,
+                "Stopped",
```

Comment:
always use lower() for string comparisons. Also, use the constant strings from codes.py rather than having strings themselves everywhere